src/protocol/bolt/src/main/java/org/apache/jmeter/protocol/bolt/sampler/BoltSampler.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.jmeter.config.ConfigTestElement;
+import org.apache.jmeter.engine.util.ConfigMergabilityIndicator;
+import org.apache.jmeter.protocol.bolt.config.BoltConnectionElement;
+import org.apache.jmeter.samplers.Entry;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.samplers.Sampler;
+import org.apache.jmeter.testbeans.TestBean;
+import org.apache.jmeter.testelement.TestElement;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.exceptions.Neo4jException;
+import org.neo4j.driver.v1.summary.ResultSummary;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+
+public class BoltSampler extends AbstractBoltTestElement implements Sampler, TestBean, ConfigMergabilityIndicator {
+
+    private static final Set<String> APPLICABLE_CONFIG_CLASSES = new HashSet<>(
+            Collections.singletonList("org.apache.jmeter.config.gui.SimpleConfigGui")); // $NON-NLS-1$
+
+    private static final ObjectReader objectMapper = new ObjectMapper().readerFor(new TypeReference<HashMap<String, Object>>() {});
+
+    @Override
+    public SampleResult sample(Entry e) {
+        SampleResult res = new SampleResult();
+        res.setSampleLabel(getName());
+        res.setSamplerData(request());
+        res.setDataType(SampleResult.TEXT);
+        res.setContentType("text/plain"); // $NON-NLS-1$
+        res.setDataEncoding(StandardCharsets.UTF_8.name());
+
+        Map<String, Object> params;
+        try {
+            params = getParamsAsMap();
+        } catch (IOException ex) {
+            return handleException(res, ex);
+        }
+
+        // Assume we will be successful
+        res.setSuccessful(true);
+        res.setResponseMessageOK();
+        res.setResponseCodeOK();
+
+        res.sampleStart();
+
+        try {
+            res.setResponseHeaders("Cypher request: " + getCypher());
+            res.setResponseData(execute(BoltConnectionElement.getDriver(), getCypher(), params), StandardCharsets.UTF_8.name());
+        } catch (Exception ex) {
+            res = handleException(res, ex);
+        } finally {
+            res.sampleEnd();
+        }
+        return res;
+    }
+
+    /**
+     * @see org.apache.jmeter.samplers.AbstractSampler#applies(org.apache.jmeter.config.ConfigTestElement)
+     */
+    @Override
+    public boolean applies(ConfigTestElement configElement) {
+        String guiClass = configElement.getProperty(TestElement.GUI_CLASS).getStringValue();
+        return APPLICABLE_CONFIG_CLASSES.contains(guiClass);
+    }
+
+    private String execute(Driver driver, String cypher, Map<String, Object> params) {
+        try (Session session = driver.session()) {
+            StatementResult statementResult = session.run(cypher, params);
+            return response(statementResult);
```

Comment:
should sampleEnd be before this or not ?

Reply:
IMO the current approach is correct, as we want to include the time taken to consume the results in the timing

Reply:
ok, just wanted to be sure. This should maybe be documented.

Reply:
Just wondering: JMeter has a concept of `latency` which typically means `time to the first byte`.

Should we use `latency` here as `the time to the first result`?